src/common.ts

```diff
@@ -99,9 +99,9 @@ export class GaxiosError<T = any> extends Error {
     message: string,
     public config: GaxiosOptionsPrepared,
     public response?: GaxiosResponse<T>,
-    public error?: Error | NodeJS.ErrnoException,
+    cause?: unknown,
```

Comment:
As `GaxiosError` is an error itself (with a few additional custom params) it would enable a chain of `Error`s to be appropriately logged along with each individual call stacks - providing a complete context for customers. The `error` param was a stop-gap, but often it would not be used properly. I think cleanly migrating would be a more optimal UX instead of both.

`unknown` is the appropriate type - cause could be anything, but should be verified through type checks.

Reply:
Perhaps a less destructive way then would be to add a deprecated tag to error, and simply continue to have it there? I'm concerned that it's a needlessly breaking change, especially since we're bundling a few other breaking changes in this release.

Reply:
A quick code search in `googleapis` will show virtually none of our libraries are using this property and it will be far more intuitive in the future to simply clean it up now.

Reply:
Right but perhaps it's being used by external users?